tests/Request/LaravelRequestTest.php

```diff
@@ -34,14 +35,25 @@ public function testCanResolveConsoleRequest()
 
     public function testCanResolveLaravelRequest()
     {
-        $resolver = new LaravelResolver($app = Mockery::mock(Application::class));
-
-        $app->shouldReceive('runningInConsole')->once()->andReturn(false);
-        $app->shouldReceive('make')->once()->with(Request::class)->andReturn($request = Mockery::mock(Request::class));
-
-        $request = $resolver->resolve();
-
-        $this->assertInstanceOf(RequestInterface::class, $request);
-        $this->assertInstanceOf(LaravelRequest::class, $request);
+        try {
+            // different versions of Laravel use slightly different mechanisms
+            // to determine if the app is 'runningInConsole'
```

Comment:
Which need each mechanism?

Reply:
Basically Laravel 5 uses this function and Laravel 6+ uses the environment variable or `PHP_SAPI` constant

There's some overlap in 5.7 & 5.8, which use the environment variable and `php_sapi_name` function